Automattic/wp-calypso
# Jetpack Connect: Auto authorize on signup via query

When creating an account, an action is dispatched:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/jetpack-connect/auth-logged-out-form.jsx#L54

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/state/jetpack-connect/actions.js#L323-L343

Which has the side effect of setting `autoAuthorize` state:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L82-L103

Which, in turn, will autoAuthorize:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/jetpack-connect/auth-logged-in-form.jsx#L101-L108

---

Rather than doing that, let's just attach a query param saying we've authorized!

`auth_approved=true`.

This PR removes a query param `new_user_started_connection ` that was apparently unused and set in an old user creation flow (@johnHackworth confirm 🙏 )

## Testing
1. Start logged out from WordPress.com
1. Make a new Jetpack site
1. Connect the site
1. Create a new user from the connection screen
1. You should be moved to the authorize step with an `auth_approved` query parameter.
1. Authorize should proceed without requiring user interaction.
